### PR TITLE
ssh, agent: add SFTP support

### PR DESCRIPTION
**Pull request by henrybarreto**

### CODE REVIEW

File: agent/sshd/server.go [hunk 1]
```diff
@@ -48,6 +49,29 @@ type Server struct {
...
+		serverOptions...,
+	)
+	if err != nil {
+		logrus.Error("unable to start SFTP subsytem: ", err)
+
+		return
+	}
+
+	if err = server.Serve(); err != nil && err != io.EOF {
+		logrus.Printf("STFP server serve error: %s\n", err)
```
Comment (otavio) [inline agent/sshd/server.go:67]: Shouldn't it use an error?

File: ssh/main.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ const (
 )
 
 func main() {
+	log.SetLevel(log.TraceLevel)
```
Comment (otavio) [inline ssh/main.go:25]: Likely a forgotten change.

**Review by otavio — CHANGES_REQUESTED** [review]

File: ssh/main.go (hunk 2 above)
Comment (henrybarreto) [reply ssh/main.go:25]: Exactly!

File: ssh/server/server.go [hunk 3]
```diff
@@ -54,43 +55,46 @@ func NewServer(opts *Options, tunnel *httptunnel.Tunnel) *Server {
...
 
-	server.sshd = &ssh.Server{ // nolint: exhaustruct
+	server.sshd = &gliderssh.Server{ // nolint: exhaustruct
 		Addr:                   opts.Addr,
 		Handler:                server.SessionHandler,
 		PasswordHandler:        handler.Password,
 		PublicKeyHandler:       handler.PublicKey,
 		SessionRequestCallback: server.sessionRequestCallback,
+		SubsystemHandlers: map[string]gliderssh.SubsystemHandler{
+			"sftp": subsystem.STFPSubsystemHandler(tunnel),
```
Comment (otavio) [inline ssh/server/server.go:65]: SFTP; check if there is other typos.

**Review by otavio — APPROVED** [review]

**Review by otavio — APPROVED** [review]

File: ssh/server/server.go [hunk 4]
```diff
@@ -13,14 +13,15 @@ import (
 	"os"
 	"time"
 
-	"github.com/gliderlabs/ssh"
+	gliderssh "github.com/gliderlabs/ssh"
```
Comment (otavio) [inline ssh/server/server.go:16]: avoid the rename.

**Review by otavio — APPROVED** [review]

**Pull request merged**